app/canvas/features/wrapper-component/wrapper-component.tsx

```diff
@@ -23,71 +26,114 @@ export const WrapperComponentDev = ({
   const { dragRefCallback, ...draggableProps } = useDraggable({
     instance,
-    isDisabled: isContentEditableDisabled === false,
+    // We can't drag if we are editing text.
+    isDisabled: isEditing === true,
   });
 
   const focusRefCallback = useEnsureFocus();
 
   const refCallback = useCallback(
     (element) => {
-      contentEditableRef.current = element;
-      // We can't drag if we are editing text.
+      if (isEditing) editableRefCallback(element);
       dragRefCallback(element);
       focusRefCallback(element);
     },
-    [contentEditableRef, dragRefCallback, focusRefCallback]
+    [dragRefCallback, focusRefCallback, editableRefCallback, isEditing]
   );
 
   const userProps = useUserProps(instance.id);
-  const readonly =
+  const readonlyProps =
     instance.component === "Input" ? { readOnly: true } : undefined;
 
   const { Component } = primitives[instance.component];
-  return (
-    <>
-      {editableToolbar}
-      <Component
-        {...userProps}
-        {...rest}
-        {...contentEditableProps}
-        {...draggableProps}
-        {...readonly}
-        className={className}
-        id={instance.id}
-        tabIndex={0}
-        data-component={instance.component}
-        data-id={instance.id}
-        ref={refCallback}
-        onClick={(event: MouseEvent) => {
-          if (instance.component === "Link") {
-            event.preventDefault();
-          }
-        }}
-        onSubmit={(event: FormEvent) => {
-          // Prevent submitting the form when clicking a button type submit
-          event.preventDefault();
+
+  const props = {
+    ...userProps,
+    ...rest,
+    ...draggableProps,
+    ...readonlyProps,
+    ...isEditingProps,
+    ...editableProps,
+    // @todo merge className with props
+    className: className,
+    // @todo stop using id to free it up to the user
+    id: instance.id,
+    tabIndex: 0,
+    "data-component": instance.component,
+    "data-id": instance.id,
+    ref: refCallback,
+    onClick: (event: MouseEvent) => {
+      // Prevent click handler for element selection
+      if (isEditing) {
+        event.stopPropagation();
+      }
+      if (instance.component === "Link") {
+        event.preventDefault();
+      }
+    },
+    onSubmit: (event: FormEvent) => {
+      // Prevent submitting the form when clicking a button type submit
+      event.preventDefault();
+    },
+  };
+
+  // Prevent rerender because in editing mode Editor controls the node.
+  const editor = useMemo(() => {
+    if (isEditing === false) return;
+    return (
+      <Editor
+        instance={instance}
+        editable={<Component {...props} />}
+        onChange={(updates) => {
+          onChangeChildren({ instanceId: instance.id, updates });
         }}
-      >
-        {renderWrapperComponentChildren(contentEditableProps.children)}
-      </Component>
-    </>
+      />
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Haven't found any other way, we need to actually prevent Editor from updating until it has finished editing, because it has to control the dom and I think its not easy to make this work otherwise.

Basically what I needed to express is that as long as isEditing is true, ignore any updates in that WrapperComponent.

Do you see a different way than this? Otherwise I will just add this explanation as inline comment

Reply:
React.memo tip from @giuseppeg has helped here